src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java

```diff
@@ -122,13 +131,39 @@ void persist(Map<String, String> eventHeaders, NotifyContextRequest notification
                     backend.createCollection(dbName, collectionName);
                 } // if
 
-                LOGGER.info("[" + this.getName() + "] Persisting data at OrionMongoSink. Database: " + dbName
-                        + ", Collection: " + collectionName + ", Data: " + recvTimeTs / 1000 + "," + recvTime + ","
-                        + entityId + "," + entityType + "," + attrName + "," + entityType + "," + attrValue + ","
-                        + attrMetadata);
-                backend.insertContextDataRaw(dbName, collectionName, recvTimeTs / 1000, recvTime,
-                        entityId, entityType, attrName, attrType, attrValue, attrMetadata);
+                if (this.rowAttrPersistence) {
+                    LOGGER.info("[" + this.getName() + "] Persisting data at OrionMongoSink. Database: "
+                            + dbName + ", Collection: " + collectionName + ", Data: " + recvTimeTs.longValue() / 1000L
+                            + "," + recvTime + "," + entityId + "," + entityType + ","
+                            + attrName + "," + attrType + "," + attrValue + "," + attrMetadata);
+                    this.backend.insertContextDataRaw(
+                            dbName, collectionName, recvTimeTs.longValue() / 1000L, recvTime,
+                            entityId, entityType, attrName, attrType, attrValue, attrMetadata);
+                } else if (this.dataModel == DataModel.COLLECTIONPERATTRIBUTE) {
+                    HashMap singleAttrs = new HashMap();
+                    singleAttrs.put(attrName, attrValue);
+                    HashMap singleMds = new HashMap();
+                    singleMds.put(attrName + "_md", attrMetadata);
+                    this.backend.insertContextDataRaw(
+                            dbName, collectionName, recvTimeTs.longValue() / 1000L, recvTime,
+                            entityId, entityType, singleAttrs, singleMds);
+                } else {
+                    attrs.put(attrName, attrValue);
+                    mds.put(attrName + "_md", attrMetadata);
+                }
             } // for
+            if (!this.rowAttrPersistence && this.dataModel != DataModel.COLLECTIONPERATTRIBUTE) {
```

Comment:
Yourselves have discovered the COLLECTIONPERATRIBUTE has no sense in column-like mode. In this case, a warn must be traced and nothis else must be done form the persistence point of view.